packages/puppeteer-core/src/node/ProductLauncher.ts

```diff
@@ -236,7 +237,14 @@ export abstract class ProductLauncher {
         await browserProcess.close();
       }
     } else {
-      await browserProcess.close();
+      try {
+        // Wait for a possible graceful shutdown.
+        await firstValueFrom(
```

Comment:
@Lightning00Blade am I holding it right?

Reply:
This will work but I would use RxJs for the whole thing. (Not sure its the better thing to do style wise)
 ```ts
// Wait for a possible graceful shutdown.
 await firstValueFrom(
  race(
    from(browserProcess.hasClosed()),
    timer(5000).pipe(
      map(
        () => {
          return from(browserProcess.close())
        } 
      )
    ) 
 )
 ```